frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+let variable0;
+// variables can be reassigned
+variable0 = "hi";
+// js is dynamically typed, so variables can also be reassigned to values with
+// different types
+variable0 = 0;
+// const variables must be initialized at declaration and cannot be reassigned
+// after declaration
+// const variable1; <- not allowed!
+const variable1 = 1;
+// variable1 = 2; <- not allowed!
+
+// variables are scoped to the block they're in. this means that they're only
+// accessible inside the curly braces ({}) they were declared.
+let outerScope = "outer!";
+// outerScope now accessible
+{
+    // outerScope still accessible
+    let innerScope = "inner!";
+    // innerScope now accessible
+}
+// innerScope no longer accessible
+// outerScope still accessible
+```
+
+Sometimes you'll want to do an operation on a variable's value and then store the result back into the variable. JavaScript has a shorthand for this:
+```js
+let foo = 100;
+// instead of writing something like this
+foo = foo + 1;
+// you can instead write this
+foo += 1;
+```
+You use this operation-then-assign syntax with any of the operations mentioned in the previous section.
+
+### Comparisons
+Comparisons let you check whether data structures satisfy some sort of condition.
+
+```js
+// double-equals (==) compares values. this yields true:
+0 == 0
+// but beware! the interpreter will try and convert types to allow for
+// comparison. this can lead to unexpected behavior like this (comparing string
+// and number), which yields true
+"0" == 0
+
+// if you want to do an equals comparison while respecting type differences,
+// you'd use a triple-equals (===) comparison.
+0 === 0 // true
+"0" === 0 // false
+
+// a similar distinction exists for the not-equals comparisons. the following
+// are equivalent, and yield false
+!(0 == 0)
+0 != "0"
+// the following are equivalent, and yield true
+!(0 === "0")
+0 !== "0"
+
+// finally, we have the ordering comparisons. these work as you would expect in
+// any other language, with the caveat that types are always converted (like
+// with ==)
+0 > 0 // false
+0 >= "0" // true
+0 < "0" // false
+0 <= 0 // true
+```
+
+JavaScript also has something called 'truthy' and 'falsy' values. Basically, all non-Boolean values will behave like `true` or `false` if you try to use them as if they were Boolean. The rule is pretty simple - all values behave like `true` except for the following, which behave like `false`:
+- `0`
+- `NaN` (a floating point not-a-number value)
+- `""` (the empty string)
+- `null`
+- `undefined`
+```js
+// we'll use bang-bang (!!) to force a truthy/falsy value to turn into a
+// boolean. it really just does a logical not (the first !) and then nots that
+// again (the second !) to get the original boolean value.
+// the following all yield `true`
+false === !!0
+false === !!""
+false === !!null
+true === !!1
+true === !!"true"
+true === !!"false"
+```
+
+### Conditionals
```

Comment:
I omitted switch statements since they're not very common. Should they be included?

Reply:
Since they're not commonly used in the language, I agree that their omission is justified.